packages/block-editor/src/components/url-input/index.js

```diff
@@ -100,6 +100,9 @@ class URLInput extends Component {
 	}
 
 	componentWillUnmount() {
+		if ( this.suggestionsRequest?.cancel ) {
+			this.suggestionsRequest.cancel();
+		}
```

Comment:
Found this during debugging, if you quickly toggle the `URLInput` before the fetch returns, it will attempt to call `setState()` on the unmounted component. This cancels the `suggestionsRequest` Promise when it unmounts to avoid the issue.

I can break it out to a separate PR if needed, the rest of the code doesn't trigger this anymore.

Reply:
Probably okay to sneak in, maybe just leave a quick note in the summary. eg "also fixes ..."

Minor: We can remove the if check using optional chaining.

```
this?.suggestionsRequest?.cancel?.()
```

See also: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining#stacking_the_optional_chaining_operator